Src/BootCamp.Chapter/CommandHandler.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace BootCamp.Chapter
+{
+    internal class CommandHandler
+    {
+        private readonly CommandArgument[] _commandText;
+        private readonly FileInfo _outputFile;
+
+        public CommandHandler(string[] commandText)
+        {
+            if (commandText is null)
+            {
+                throw new ArgumentNullException(nameof(commandText));
+            }
+
+            if (commandText.Any(x => string.IsNullOrEmpty(x)))
+            {
+                throw new ArgumentException("Unexpected empty or null argument.", nameof(commandText));
+            }
+
+            if (OutputFileArgumentProvided(commandText))
+            {
+                _outputFile = new FileInfo(commandText[1]);
+            }
+
+            _commandText = commandText[0].Split(' ').Select(x => new CommandArgument(x)).ToArray();
+        }
+
+        private static bool OutputFileArgumentProvided(string[] commandText)
+        {
+            return commandText.Length == 2;
+        }
+
+        private bool CommandHasArguments() => _commandText.Length >= 2;
+
+        internal Action<Stream> ParseCommand()
+        {
+            var cmd = _commandText[0].NormalizedArgument;
+
+            if (cmd == "CITY")
+            {
+                return HandleCityCommand();
+            }
+
+            if (cmd == "TIME")
+            {
+                return HandleTimeCommand();
+            }
+
+            if (cmd == "DAILY")
+            {
+                return HandleDailyCommand();
+            }
+
+            if (cmd == "FULL")
+            {
+                return HandleFullCommand();
+            }
+
+            throw new InvalidCommandException();
+        }
+
+        private Action<Stream> HandleFullCommand()
+        {
+            if (CommandHasArguments())
+            {
+                throw new InvalidCommandException();
+            }
+
+            if (_outputFile != null)
+            {
+                throw new InvalidCommandException();
+            }
+
+            var command = new FullCommand();
+            command.FilterCompleted += (s, e) => OutputShopReport(e.Shop);
+            return command.ExecuteCommand;
+        }
+
+        private void OutputShopReport(IGrouping<string, Transaction> shop)
+        {
+            using var outputText = File.CreateText($"{shop.Key}.csv");
+            outputText.WriteLine("City, Street, Item, DateTime, Price");
+            foreach (var transaction in shop)
```

Comment:
For loop, it's better to use StringBuilder to concatinate the strings, then write all that to output. 10s of times more performant.